src/service/github.service.ts

```diff
@@ -128,14 +130,26 @@ export class GitHubService {
   public async ReadGist(
     GIST: string
   ): Promise<GitHubApi.Response<IFixGistResponse>> {
-    return await this.github.gists.get({ gist_id: GIST });
+    const promise = this.github.gists.get({ gist_id: GIST });
+    const res = await promise.catch(err => {

```

Comment:
How are we handling the exceptions, as `await` will throw exception directly to driver functions.
By handling exceptions in the `catch` all of the exceptions wont work in that code where `ReadGist` is being called?

What are your thoughs

Reply:
If a promise is caught using `catch`, we can examine the error and display an appropriate error to the user. If the error is not caught, then it will be thrown automatically in `ReadGist`. 

We are also awaiting the `catch` function which will return `undefined` if an error occurs instead of throwing the error.